test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1158,6 +1158,8 @@ UCS_TEST_P(test_ucp_sockaddr_protocols, tag_zcopy_64k_exp_sync,
 
 UCS_TEST_P(test_ucp_sockaddr_protocols, tag_rndv_exp, "RNDV_THRESH=10k")
 {
+    /* run send_recv twice so that the rndv threshold would be used */
```

Comment:
what does it mean?

Reply:
when the rndv_thresh is set, it's not used for the first message, so even if the message is larger than the set rndv_thresh, rndv will not be selected. it will be selected on the second message.

Reply:
maybe send smaller message as a first one?

Reply:
@brminich why?

Reply:
> maybe send smaller message as a first one?

@brminich initially this test was added to check that **first large**  message works correctly

Reply:
what is meant by correctly? That it is correctly sent, even though using eager protocol?

Reply:
yes, aka ep/request re-configuration issue due to protocol mismatch

Reply:
@evgeny-leksikov the 'first large message' will be sent with rndv if it's not the first message sent (before the connection was established). so sending a small message first shouldn't contradict what you are saying.

Reply:
AFAIU this is exactly **before** wireup is finished case

Reply:
let's change "test_tag_send_recv" to run several iterations (e.g 10) and add this comment there:
`/* send multiple messages to test the protocol both before and after connection establishment */`